From 991a3383c2166c719bd94be53734998c3664352f Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Thu, 31 Jan 2019 17:16:39 -0500
Subject: [PATCH 49/51] ARM: dts: am33xx: add device tree entry for SGX

commit fc66dfef3e7020afe2cd5e58db5607f012bc620b from
https://git.ti.com/cgit/processor-sdk/processor-sdk-linux/?h=processor-sdk-linux-01.00.00

SGX device entry is added to base am33xx dtsi file.

Note: The SGX is disabled by default because it is an optional module
and only some AM335x variants contain this module, such as AM3358 and
AM3357.  The status should be overwritten as "OK" at the corresponding
board.dts

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
Signed-off-by: Anand Balagopalakrishnan <anandb@ti.com>
[praneeth@ti.com: rebase from graphics-ti-linux-4.14.y]
Signed-off-by: Praneeth Bajjuri <praneeth@ti.com>
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 arch/arm/boot/dts/am33xx.dtsi | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index e99bd687af34..2c9bd999ea0e 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -558,6 +558,16 @@
 			       <&edma 5 0>;
 			dma-names = "tx", "rx";
 		};
+
+		sgx: sgx@56000000 {
+                        compatible = "ti,am3352-sgx530", "img,sgx530";
+                        ti,hwmods = "gfx";
+                        reg = <0x56000000 0x10000>;
+                        interrupts = <37>;
+                        clocks = <&gfx_fck_div_ck>;
+                        clock-names = "fclk";
+                        status = "disabled";
+                };
 	};
 };
 
-- 
2.17.1

